cpp/src/arrow/util/async_util.cc

```diff
@@ -306,7 +313,10 @@ class ThrottledAsyncTaskSchedulerImpl
     std::optional<Future<>> maybe_backoff = throttle_->TryAcquire(latched_cost);
     if (maybe_backoff) {
 #ifdef ARROW_WITH_OPENTELEMETRY
-      TraceTaskQueued(task.get(), span());
+      EVENT(span(), "Task submission throttled",
+            {{"task.name", ::opentelemetry::nostd::string_view(task->name().data(),
+                                                               task->name().size())},
+             {"task.cost", task->cost()}});
```

Comment:
`span()` is the scheduler's span.  This puts an event on the scheduler's span that submission was throttled.  I don't think that's terribly interesting.

What I was aiming for with `TraceTaskQueued` was to actually start the task's span.  Then, later, in `TraceTaskSubmitted`, it would register a "task submitted" event.

So a throttled task (that is queued for 1 time unit and takes 2 time units to run) would look something like this...

```
T+0 Task Span Begins
T+1 Task Submitted
T+3 Task Ends
```

A non-throttled task would look something like:

```
T+0 Task Begins
T+2 Task Ends
```